From 0f5c633c5bab755eb76aa0f65244de5681233a56 Mon Sep 17 00:00:00 2001
From: Jasem Mutlaq <mutlaqja@ikarustech.com>
Date: Thu, 20 Jan 2022 11:11:41 +0000
Subject: [PATCH] Update sequence prefix for ADU-calculated flat frames since
 full prefix can...

Update sequence prefix for ADU-calculated flat frames since full prefix can change due to dynamic exposure calculated in the process
---
 kstars/ekos/capture/capture.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/kstars/ekos/capture/capture.cpp b/kstars/ekos/capture/capture.cpp
index 6405e7fec..bb4ddffb5 100644
--- a/kstars/ekos/capture/capture.cpp
+++ b/kstars/ekos/capture/capture.cpp
@@ -6694,6 +6694,14 @@ bool Capture::processPostCaptureCalibrationStage()
                     placeholderPath.processJobInfo(activeJob, activeJob->getCoreProperty(SequenceJob::SJ_TargetName).toString());
                     // Mark calibration as complete
                     calibrationStage = CAL_CALIBRATION_COMPLETE;
+
+                    // Must update sequence prefix as this step is only done in prepareJob
+                    // but since the duration has now been updated, we must take care to update signature
+                    // since it may include a placeholder for duration which would affect it.
+                    if (currentCCD->getUploadMode() != ISD::CCD::UPLOAD_LOCAL)
+                        updateSequencePrefix(activeJob->getCoreProperty(SequenceJob::SJ_FullPrefix).toString(),
+                                             QFileInfo(activeJob->getSignature()).path());
+
                     startNextExposure();
                     return false;
                 }
-- 
GitLab

